scripts/convert_yaml_to_sdf.py

```diff
@@ -0,0 +1,368 @@
+
+
+def read_areas(areas, link_names):
+    """
+    Convert areas to links with a virtual area
+    :param areas: list of areas
+    :type areas: list
+    :param link_names: list of link names already used
+    :type link_names: list
+    :return: list of links with a virtual area child element
+    :rtype: list
+    """
+    sdf_link = []
+    if not isinstance(areas, list):
+        print("areas should be a list")
+        return sdf_link
+
+    area_names = []
+    for area in areas:
+        name = area["name"]
+        if name == "near":
+            continue
+
+        uname = unique_name(name, link_names)
+        if not uname == name:
+            print("Name of area has changed from '{}' to '{}'".format(name, uname))
+        sdf_link_item = {"name": uname}
+
+        for shape_item in area["shape"]:
+            geometry, super_pose = read_geometry(shape_item)
+            shape_name = unique_name(uname, area_names)
+            sdf_link_item["virtual_area"] = {"name": shape_name, "geometry": geometry}
+            if super_pose:
+                sdf_link_item["virtual_area"]["pose"] = super_pose
+
+            sdf_link.append(sdf_link_item)
+
+    return sdf_link
+
+
+def parse_to_xml(xml, item, list_name=""):
+    """
+    Extend XML with elements from a dict, list or a string
+    :param xml: xml element
+    :type xml: xml.etree.ElementTree.Element
+    :param item: dict, list or str
+    :type item: dict or list or str
+    :param list_name: name of list, needs to be passed on by parent for each element
+    :type list_name: str
+    """
+    if isinstance(item, list):
+        if not list_name:
+            print("list_name should be passed on by parent in case of a list")
+            return -1
+        for v in item:
+            child = ET.SubElement(xml, list_name)
+            parse_to_xml(child, v)
+    elif isinstance(item, dict):
+        for k, v in item.items():
+            # attributes
+            if k == "name" or k == "version":
+                xml.set(k, v)
+                continue
+
+            if isinstance(v, list):
+                parse_to_xml(xml, v, k)
+            else:
+                child = ET.SubElement(xml, k)
+                parse_to_xml(child, v)
+    elif isinstance(item, str):
+        xml.text = item
+    else:
+        print("Cannot not parse object type: '{}'".format(type(item)))
+
+
+def write_xml_to_file(xml_element, path):
+    """
+    write xml element to a file
+    :param xml_element: xml element
+    :type xml_element: xml.etree.ElementTree.Element
+    :param path: full path of file, which to write
+    :type path: str
+    """
+    # generate xml string with indentation
+    xml_dom = parseString(ET.tostring(xml_element, encoding="utf-8"))
+    pretty_xml_string = xml_dom.toprettyxml(indent="  ")
+
+    with open(path, "w") as f:
+        f.write(pretty_xml_string)
+
+
+def main(model_name):
+    # get model path
+    model_path = get_model_path(model, "yaml")
```

Comment:
Should this be
```suggestion
    model_path = get_model_path(model_name, "yaml")
```
perhaps?